### ISSUE TITLE
[FEATURE] Replace malwaredomain list

### ISSUE THREAD

**Issue by ajinabraham** [MobSF/Mobile-Security-Framework-MobSF#2472]
If you're requesting a new feature/enhancement, explain why you'd like it to be added and it's importance.

**Is your feature request related to a problem? Please describe.**
A possible alternative: https://gitlab.com/malware-filter/urlhaus-filter

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by alanhasn** [comment]
Looked into this. The existing `malware_check()` path (`MalwareDomainCheck.py`) is effectively dead weight today:

- `MALWARE_DB_URL` (`https://www.malwaredomainlist.com/mdlcsv.php`) is defined in settings but never referenced anywhere else in the codebase - there's no code that fetches or refreshes it.
- The bundled `mobsf/signatures/malwaredomainlist` file is a static snapshot whose oldest entries date back to **2009**, with no update mechanism (unlike `maltrail_check()` / `update_maltrail_db()`, which does self-update from a live source).
- malwaredomainlist.com itself has been defunct for years, so even fixing `MALWARE_DB_URL` wouldn't help.

So `malware_check()` currently just runs against 16-year-old stale data every scan without anyone noticing, while `maltrail_check()` is the only check actually doing real work.

Two ways to go, and it'd help to know which direction you'd prefer before someone puts a PR together:

1. **Just remove the dead path** - drop `malware_check()`, the static file, and the unused `MALWARE_DB_URL` setting, and keep `maltrail_check()` as the sole (working) domain check. Small, safe cleanup, no new dependency.
2. **Replace it with a real live source** - integrate something like [urlhaus-filter](https://gitlab.com/malware-filter/urlhaus-filter) (as suggested above) as a second, actively-maintained check alongside maltrail, with its own update mechanism mirroring `update_maltrail_db()`.

Happy to send a PR for either, depending on what you'd rather have.

**Comment by ajinabraham** [comment]
I agree with both. We can deprecate and remove the logic related to `www.malwaredomainlist.com` and possibly add URLhaus.

The thing with URLhaus is that it covers more than just malware checks, so we may need to define what exactly we want to cover and potentially update the UI beyond malware checks for domains if we decide to go down that path. I wanted to draft a plan first before implementing URLhaus to make sure it aligns with the project.

**Comment by alanhasn** [comment]
Sounds good - opened #2687 scoped to just the part we agreed on (dropping the dead malwaredomainlist.com path), nothing URLhaus-related in it. Makes sense to hold off on that until you've had a chance to draft the plan, since it sounds like it could touch more than just the domain-malware check. Happy to help once that direction is clearer.
